File: Sources/honor-ed-video-adapters/honor-ed-video-adapters.swift
import Foundation

public enum EntryPoint {
    public static let playerUrl = Bundle.module.url(forResource: "HonorPlayer", withExtension: "html")
}

public enum HEVideoPlayerState: Int, Equatable {
    case unstarted = 0
    case ended = 1
    case playing = 2
    case paused = 3
    case buffering = 4
}

public enum HEVideoPlayerError: Int, Error {
    case playerError = 1
    case notFound = 2
    case notInitialized = 3
    case apiLoadError = 4
    case adaptorLayerError = 5
    case invalidPermissions = 6
    case unknown = -1
}

File: kotlin/videoadaptor/src/main/java/honor/education/videoadaptor/honor_ed_video_adapters.kt
package honor.education.videoadaptor

import android.content.Context

public object EntryPoint {
    // These will refer to assets, since there's no Bundle.module in Android
    const val playerUrl: String = "file:///android_asset/HonorPlayer.html"
    const val playerJsUrl: String = "file:///android_asset/main.js"
}

public enum class HEVideoPlayerState(val value: Int) {
    UNSTARTED(0),
    ENDED(1),
    PLAYING(2),
    PAUSED(3),
    BUFFERING(4);

    companion object {
        fun from(value: Int): HEVideoPlayerState? = values().find { it.value == value }
    }
}

public enum class HEVideoPlayerError(val code: Int) {
    PLAYER_ERROR(1),
    NOT_FOUND(2),
    NOT_INITIALIZED(3),
    API_LOAD_ERROR(4),
    ADAPTOR_LAYER_ERROR(5),
    INVALID_PERMISSIONS(6),
    UNKNOWN(-1);

    companion object {
        fun from(code: Int): HEVideoPlayerError? = values().find { it.code == code }
    }
}